backend/infrahub/cli/db.py

```diff
@@ -185,98 +192,109 @@ async def update_core_schema(  # pylint: disable=too-many-statements
-        )
-        if error_messages:
-            rprint(f"{error_badge} | Unable to update the schema, due to failed validations")
-            for message in error_messages:
-                rprint(message)
-            raise typer.Exit(1)
-
-        # ----------------------------------------------------------
-        # Update the schema
-        # ----------------------------------------------------------
-        origin_schema = branch_schema.duplicate()
-
-        # Update the internal schema
-        schema_default_branch.load_schema(schema=SchemaRoot(**internal_schema))
-        schema_default_branch.process()
-        registry.schema.set_schema_branch(name=default_branch.name, schema=schema_default_branch)
-
-        async with db.start_transaction() as dbt:
-            await registry.schema.update_schema_branch(
-                schema=candidate_schema,
-                db=dbt,
-                branch=default_branch.name,
-                diff=result.diff,
-                limit=result.diff.all,
-                update_db=True,
+        with prefect_test_harness():
+            # ----------------------------------------------------------
+            # Initialize Schema and Registry
+            # ----------------------------------------------------------
+            service = InfrahubServices(
+                database=db, message_bus=BusSimulator(database=db), workflow=WorkflowLocalExecution()
             )
-            default_branch.update_schema_hash()
-            rprint("The Core Schema has been updated")
+            await initialize_registry(db=db)
+
+            default_branch = registry.get_branch_from_registry(branch=registry.default_branch)
+
+            registry.schema = SchemaManager()
+            schema = SchemaRoot(**internal_schema)
+            registry.schema.register_schema(schema=schema)
+
+            # ----------------------------------------------------------
+            # Load Current Schema from the database
+            # ----------------------------------------------------------
+            schema_default_branch = await registry.schema.load_schema_from_db(db=db, branch=default_branch)
+            registry.schema.set_schema_branch(name=default_branch.name, schema=schema_default_branch)
+            branch_schema = registry.schema.get_schema_branch(name=registry.default_branch)
+
+            candidate_schema = branch_schema.duplicate()
+            candidate_schema.load_schema(schema=SchemaRoot(**internal_schema))
+            candidate_schema.load_schema(schema=SchemaRoot(**core_models))
+            candidate_schema.load_schema(schema=SchemaRoot(**deprecated_models))
+            candidate_schema.process()
+
+            result = branch_schema.validate_update(other=candidate_schema, enforce_update_support=False)
+            if result.errors:
+                rprint(f"{error_badge} | Unable to update the schema, due to failed validations")
```

Comment:
Agreed, I think it would make sense to have all the logic executed in this command to be wrap inside a prefect flow, this way the print statement will be logged and we could easily capture everything in prefect
I'll open an issue to track this task